### ISSUE TITLE
Installation issue: tcl

### ISSUE THREAD

**Issue by alexispaz** [spack/spack#10874]
### Steps to reproduce the issue

```console
$ spack install tcl

==> Error: IOError: [Errno 2] No such file or directory: '/home/alexis/spack/opt/spack/linux-opensuse42-x86_64/gcc-8.3.0/tcl-8.6.6-wqqbl4jhje4xuyszmppxtjgki4qgkuqa/lib/tclConfig.sh'

/home/alexis/spack/var/spack/repos/builtin/packages/tcl/package.py:58, in install:
         55            # Replace stage dir -> installed src dir in tclConfig
         56            filter_file(
         57                stage_src, installed_src,
  >>     58                join_path(self.spec.prefix, 'lib', 'tclConfig.sh'))
         59
         60        # Don't install binaries in src/ tree
         61        with working_dir(join_path(installed_src, self.configure_directory)):

See build log for details:
  /home/alexis/spack/var/spack/stage/tcl-8.6.6-wqqbl4jhje4xuyszmppxtjgki4qgkuqa/tcl8.6.6/spack-build.out
```
and the last lines of build.out:

```console
==> [2019-03-12-16:41:19.923622] 'make' '-j4' 'install-private-headers'
Installing private header files to /home/alexis/spack/opt/spack/linux-opensuse42-x86_64/gcc-8.3.0/tcl-8.6.6-wqqbl4jhje4xuyszmppxtjgki4qgkuqa/include/
==> [2019-03-12-16:41:19.971745] Installing /home/alexis/spack/var/spack/stage/tcl-8.6.6-wqqbl4jhje4xuyszmppxtjgki4qgkuqa/tcl8.6.6 to /home/alexis/spack/opt/spack/linux-opensuse42-x86_64/gcc-8.3.0/tcl-8.6.6-wqqbl4jhje4xuyszmppxtjgki4qgkuqa/share/tcl/src
==> [2019-03-12-16:41:20.219371] FILTER FILE: /home/alexis/spack/opt/spack/linux-opensuse42-x86_64/gcc-8.3.0/tcl-8.6.6-wqqbl4jhje4xuyszmppxtjgki4qgkuqa/lib/tclConfig.sh [replacing "/home/alexis/spack/var/spack/stage/tcl-8.6.6-wqqbl4jhje4xuyszmppxtjgki4qgkuqa/tcl8.6.6"]
```

### Platform and user environment

Please report your OS here:
```commandline
$ uname -a
Linux greenline 4.4.90-18.32-default #1 SMP Fri Oct 6 13:30:08 UTC 2017 (465b410) x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -d
Description:    openSUSE Leap 42.2
```

**Comment by alexispaz** [comment]
It seems that after installing the source, the package search for `tclConfig.sh` in a wrong address? This patch solve my problem:

```
b/var/spack/repos/builtin/packages/tcl/package.py
index d5bd68674..5e3ceb341 100644
--- a/var/spack/repos/builtin/packages/tcl/package.py
+++ b/var/spack/repos/builtin/packages/tcl/package.py
@@ -55,7 +55,7 @@ class Tcl(AutotoolsPackage):
             # Replace stage dir -> installed src dir in tclConfig
             filter_file(
                 stage_src, installed_src,
-                join_path(self.spec.prefix, 'lib', 'tclConfig.sh'))
+                join_path(installed_src, 'unix', 'tclConfig.sh'))
 
         # Don't install binaries in src/ tree
         with working_dir(join_path(installed_src, self.configure_directory)):
```

But I guess somebody has to check if this is the solution (I am just a newby user).

**Comment by asemin** [comment]
I've got the same problem on openSUSE Leap 15.1, and the proposed patch solved the issue.

**Comment by dh4nav** [comment]
I'm having this problem too, and I'd like to argue that it's actually high impact, because tcl is a dependency in the installation of lmod, which in turn is part of the modules config tutorial. So every new user has a high chance of running into this issue.

**Comment by xdelaruelle** [comment]
@adamjstewart @alalazo Maybe this issue has been fixed by #19761. At least, 4fcb7d1c502c4a7ddcbe5cf8017f02367e32935f has updated the problematic code line this issue is about.
